Sources/SwiftKueryPostgreSQL/PostgreSQLConnection.swift

```diff
@@ -220,138 +235,157 @@ public class PostgreSQLConnection: Connection {
+            let status = PQresultStatus(result)
+            if status != PGRES_COMMAND_OK {
+                self.setState(.idle)
+                var errorMessage = "Failed to create prepared statement."
+                if let error = String(validatingUTF8: PQerrorMessage(self.connection)) {
+                    errorMessage += " Error: \(error)."
+                }
+                PQclear(result)
+                return self.runCompletionHandler(.error(QueryError.databaseError(errorMessage)), onCompletion: onCompletion)
             }
+            self.setState(.idle)
             PQclear(result)
-            return errorMessage
+            self.preparedStatements.insert(statementName)
+            return self.runCompletionHandler(.success(PostgreSQLPreparedStatement(name: statementName, query: raw)), onCompletion: onCompletion)
         }
-        setState(.idle)
-        PQclear(result)
-        preparedStatements.insert(name)
-        return nil
     }
 
     /// Execute a prepared statement.
     ///
     /// - Parameter preparedStatement: The prepared statement to execute.
     /// - Parameter onCompletion: The function to be called when the execution has completed.
     public func execute(preparedStatement: PreparedStatement, onCompletion: @escaping ((QueryResult) -> ()))  {
-        execute(query: nil, preparedStatement: preparedStatement, with: [Any?](), onCompletion: onCompletion)
+        DispatchQueue.global().async {
+            self.execute(query: nil, preparedStatement: preparedStatement, with: [Any?](), onCompletion: onCompletion)
+        }
     }
-    
+
     /// Execute a prepared statement with parameters.
     ///
     /// - Parameter preparedStatement: The prepared statement to execute.
     /// - Parameter parameters: An array of the parameters.
     /// - Parameter onCompletion: The function to be called when the execution has completed.
     public func execute(preparedStatement: PreparedStatement, parameters: [Any?], onCompletion: @escaping ((QueryResult) -> ())) {
-        execute(query: nil, preparedStatement: preparedStatement, with: parameters, onCompletion: onCompletion)
+        DispatchQueue.global().async {
+            self.execute(query: nil, preparedStatement: preparedStatement, with: parameters, onCompletion: onCompletion)
+        }
     }
-    
+
     /// Execute a prepared statement with parameters.
     ///
     /// - Parameter preparedStatement: The prepared statement to execute.
     /// - Parameter parameters: A dictionary of the parameters with parameter names as the keys.
     /// - Parameter onCompletion: The function to be called when the execution has completed.
     public func execute(preparedStatement: PreparedStatement, parameters: [String:Any?], onCompletion: @escaping ((QueryResult) -> ())) {
-        onCompletion(.error(QueryError.unsupported("Named parameters in prepared statemennts are not supported in PostgreSQL")))
+        DispatchQueue.global().async {
```

Comment:
I don't think it is necessary to offload the completion handler for this error case.  Also, there's a typo in the message (`statemennts`).